ls: compute the correct exit code

BenWiederhake opened the pull request:
Turns out, `ls`'s exit code is actually quite complex, and convincing clap to do what we need is not trivial.

#6162 grew too much, and in particular `ls` is a big hassle. So let's deal with `ls` in isolation first, and I'll rebase #6162 once this PR is merged.

sylvestre commented on src/uu/ls/src/ls.rs:
```diff
@@ -1150,7 +1156,18 @@ impl Config {
     let command = uu_app();
 
-    let matches = command.try_get_matches_from(args)?;
+    let matches = match command.try_get_matches_from(args) {
```
maybe document this a bit more

BenWiederhake commented:
Changes since last push:
- Rebased on current main
- Added a short explanation how each `command.try_get_matches_from` match-arm is reached; thanks @sylvestre!
